var/spack/repos/builtin/packages/ibm-essl/package.py

```diff
@@ -0,0 +1,92 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class IbmEssl(Package):
+    """Engineering and Scientific Subroutine Library (ESSL)."""
+
+    homepage = "https://www.ibm.com/systems/power/software/essl/"
+    url      = "ibm-essl"
+
+    variant('ilp64', default=False, description='64 bit integers')
+    variant(
+        'threads', default='openmp',
+        description='Multithreading support',
+        values=('openmp', 'none'),
+        multi=False
+    )
+    variant('cuda', default=False, description='CUDA acceleration')
+
+    provides('blas')
```

Comment:
i think you should also add
```
provides('lapack')
provides('scalapack')
```
and fill those libs accordingly (look at `intel-mkl` for example).

Reply:
never-mind, looks like i mistakenly thought that `essl` does everything (like `mkl`), but it looks like there is a separate `parallel essl` for scalapack.